ktor-shared/ktor-websocket-serialization/common/src/WebsocketChannelSerialization.kt

```diff
@@ -0,0 +1,78 @@
+import io.ktor.http.cio.websocket.*
+import io.ktor.serialization.*
+import io.ktor.util.reflect.*
+import io.ktor.utils.io.charsets.*
+import kotlinx.coroutines.channels.*
+
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+public object WebsocketChannelSerialization {
+    /**
+     * Serializes [data] to a frame and enqueues this frame.
+     * May suspend if the [outgoing] queue is full.
+     * If the [outgoing] channel is already closed, throws an exception, so it is impossible to transfer any message.
+     * Frames sent after a Close frame are silently ignored.
+     * Note that a Close frame could be sent automatically in reply to a peer's Close frame unless it is a raw WebSocket session.
+     *
+     * @param data The data to serialize
+     * @param converter The WebSocket converter
+     * @param charset Response charset
+     * @param outgoing An outgoing WebSocket send channel
+     */
+    public suspend inline fun <reified T : Any> sendSerialized(
```

Comment:
maybe make it top level function and add `WebSocketSession` as a receiver to limit its scope?